From bb058c857fb30f942c88add9d0404c3d6f57f231 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Wed, 3 Feb 2021 14:58:18 +0800
Subject: [PATCH] aufs: remove the redundant fput function invoking

This issue is introduced by commit bad361d963fb("aufs5:
aufs5-mmap"). The fput should be placed by vma_fput,
but it is missed. So, remove the redundant fput with this patch.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 mm/mmap.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mm/mmap.c b/mm/mmap.c
index d87fcdc32c71..09b10e20c097 100644
--- a/mm/mmap.c
+++ b/mm/mmap.c
@@ -1897,7 +1897,6 @@ unsigned long mmap_region(struct file *file, unsigned long addr,
 	return addr;
 
 unmap_and_free_vma:
-	fput(vma->vm_file);
 	vma_fput(vma);
 	vma->vm_file = NULL;
 
-- 
2.19.1

